### DISCUSSION TITLE
Problem upgrading from 110.0.280 -> 112.2.70 or higher

### DISCUSSION THREAD

**Discussion by rocksdanister** [cefsharp/CefSharp#4618] [General]
**What version are you using?**
111.2.70

**Are you using WPF/WinForms/OffScreen?**
Winforms

**Are you targeting .Net Core? If yes which version?**
No, .NET Framework 4.7.2

Upgrading the project from 110.0.280 -> 111.2.70 is successful.
Upgrading from 111.2.70 or 110.0.280 -> 112.2.70 or higher results in runtime error:
```
[1028/094731.929:WARNING:policy_logger.cc(148)] :components\enterprise\browser\controller\chrome_browser_cloud_management_controller.cc(88) Could not create policy manager as CBCM is not enabled.
[1028/094732.075:ERROR:network_service_instance_impl.cc(663)] Network service crashed, restarting service.
[1028/094732.119:ERROR:network_service_instance_impl.cc(663)] Network service crashed, restarting service.
[1028/094732.162:ERROR:network_service_instance_impl.cc(663)] Network service crashed, restarting service.
[1028/094732.205:ERROR:network_service_instance_impl.cc(663)] Network service crashed, restarting service.
[1028/094732.248:ERROR:network_service_instance_impl.cc(663)] Network service crashed, restarting service.
[1028/094732.289:ERROR:network_service_instance_impl.cc(663)] Network service crashed, restarting service.
[1028/094732.330:ERROR:network_service_instance_impl.cc(663)] Network service crashed, restarting service.
```

**Minimal Example**
https://github.com/cefsharp/CefSharp.MinimalExample

Unable to reproduce it so far.

**Source**
https://github.com/rocksdanister/lively-cef

**Comment by amaitland** [comment]
> **Source**
> https://github.com/rocksdanister/lively-cef

What's required to run the app in debug? e.g Load a webpage like google.

**Comment by rocksdanister** [comment]
https://github.com/rocksdanister/lively-cef/blob/8981ddca64a38d5d9d3af769048417dfd9960255/src/Lively.PlayerCefSharp/Form1.cs#L51-L59

```cs
startArgs = new StartArgs
    {
        // .html fullpath
        Url = @"http://google.com/",
        //online or local(file)
        Type = "online",
        // LivelyProperties.json path if any
        Properties = @"",
        SysInfo = false,
        NowPlaying = false,
        AudioVisualizer = false,
    };
```

**Reply by amaitland** [reply]
Thanks. I can reproduce the problem, will have to look deeper into what's going on.

**Reply by amaitland** [reply]
This was a very strange one. Finally think I understand what's going on. `Chromium` uses the `--type` command line arg internally, so that's screwing up the sub processes somehow (not exactly sure how).

In the startup command line args there is the following:

```
--url "C:\Users\rocks\Documents\Lively Wallpaper Works\Heartfelt ST\index.html" --type local --display shit
```

If you delete the `--type local` from the args then the browser will load successfully (given the startup args you provided earlier).

So you'll probably have to change from using `--type` to something else.

**Reply by amaitland** [reply]
Had another though, you could try setting https://cefsharp.github.io/api/118.6.x/html/P_CefSharp_CefSettingsBase_CommandLineArgsDisabled.htm to tell chromium not to parse the global command line.

**Reply by rocksdanister** [reply]
Thanks, I forgot about the command line args set in visual studio.. like you said its weird this issue only started happening now.

Regarding `CommandLineArgsDisable` property based on what I understand won't it disable `settings.CefCommandLineArgs.Add` if some of the argument is for the process?

**Reply by amaitland** [reply]
> Regarding `CommandLineArgsDisable` property based on what I understand won't it disable `settings.CefCommandLineArgs.Add` if some of the argument is for the process?

It should only disable parsing of the global command line, args added to `CefCommandLineArgs` should still work as they're added directly.

Quick test and it appears to be working as expected. Been a long time since I tested `CommandLineArgsDisable`, so if you do go down that path just double check the args you are passed as expected either via task manager or loading `chrome://version`.

I'd suggest removing the `--` from your command line args that are set programatically.

Instead of
```c#
//Creates GPUCache regardless even if disk CachePath is not set!
settings.CefCommandLineArgs.Add("--disable-gpu-shader-disk-cache");
```

Use:
```c#
//Creates GPUCache regardless even if disk CachePath is not set!
settings.CefCommandLineArgs.Add("disable-gpu-shader-disk-cache");
```

That `--` is for when they're set via the actual command line.

**Reply by rocksdanister** [reply]
Thanks for the suggestions.

I do plan to change `--type` and remove `CommandLineArgsDisable`, have you found out why Cef process is taking the global commandline arguments? my concern would be in the future this bug will surface again if Cef/my project add a new flag.

**Reply by amaitland** [reply]
> I do plan to change `--type` and remove `CommandLineArgsDisable`

Are you seeing issues having it disabled? I'd probably just leave it disabled unless there's a specific problem.

> my concern would be in the future this bug will surface again if Cef/my project add a new flag.

If you want to go down this path then the safest option would be to prefix your command line args. e.g. `--lively-type`

> have you found out why Cef process is taking the global commandline arguments?

Needle in a haystack

**Reply by rocksdanister** [reply]
Things are working fine on my testing but just to be sure I'll also update the commands, this can be closed.. thanks.

Just a heads up you may want to add about Chromium started using PerMonitorV2 scaling in release notes.
https://github.com/rocksdanister/lively/issues/1996
